### PR TITLE
:sparkles: feat(Server): Implement function handler try catch

### LINKED ISSUE #59: Server: Create function handle try catch in controller

### PR DESCRIPTION
**Pull request by trungpham71198**
Closes #59

### CODE REVIEW

File: apps/server/src/datasources/controllers/user/mutation.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
-import { Request, Response, NextFunction } from 'express';
+import { Request, Response } from 'express';
 
-export const signUp = async (req: Request, res: Response, next: NextFunction): Promise<Response> => {
-  try {
-    return res.status(200).json(req.body);
-  } catch (error) {
-    next(error);
+import { useHandler } from '../../utils';
```
Comment (khoaquocngo) [inline apps/server/src/datasources/controllers/user/mutation.ts:3]: useHandler to global.useHandler. How do you think?

Comment (gialynguyen) [reply apps/server/src/datasources/controllers/user/mutation.ts:3]: I don't think this is a good idea. Instead, we could config the alias path to make the import paths brief.

File: apps/server/src/datasources/utils/useHandler.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Request, Response, NextFunction } from 'express';
+
+/**
+ * It takes a function that returns a promise, and returns a function that returns a promise
+ * @param fn - (req: Request, res: Response, next: NextFunction) => Promise<Response>
+ * @returns A function that takes in a request, response, and next function.
+ */
+export function useHandler(fn: (req: Request, res: Response, next: NextFunction) => Promise<Response>) {
```
Comment (gialynguyen) [inline apps/server/src/datasources/utils/useHandler.ts:8]: Its name should be `useHttpHandler` to be more meaningful.

**Review by gialynguyen — CHANGES_REQUESTED** [review]

**Comment by gialynguyen** [comment]
Blocked by #64

**Review by khoaquocngo — APPROVED** [review]

**Comment by trungpham71198** [comment]
> 

@gialynguyen done! Blocked by #64

**Review by gialynguyen — APPROVED** [review]

**Pull request merged**